packages/alpinejs/src/utils/error.js

```diff
@@ -0,0 +1,17 @@
+import {dispatch} from "./dispatch";
+
+export function tryCatch(el, expression, callback, ...args) {
+    try {
+        return callback(...args)
+    } catch (e) {
+        handleError( e, el, expression )
+    }
+}
+
+export function handleError(error, el, expression = undefined) {
+    Object.assign( error, { el, expression } )
+
+    console.warn(`Alpine Expression Error: ${error.message}\n\n${ expression ? 'Expression: \"' + expression + '\"\n\n' : '' }`, el)
+
+    dispatch( document, "alpine:error", { error } )
```

Comment:
We probably want to pass the element and expression in here (if they're set)

Reply:
Okay, we can add that. They were getting assigned to the error but I agree

Reply:
We updated it @HugoDF to be off error object and on main detail object if you want to look again